src/main/kotlin/gg/essential/elementa/utils/text.kt

```diff
@@ -23,9 +23,9 @@ fun getStringSplitToWidthTruncated(
     val suffixWidth = trimmedTextSuffix.width(textScale,fontProvider)
 
     return lines.subList(0, maxLines).mapIndexed { index, contents ->
-        if (index == maxLines - 1) {
-            var length = contents.lastIndex
-            while (contents.substring(0, length).width(textScale,fontProvider) + suffixWidth > maxLineWidth * textScale)
+        var length = contents.lastIndex
+        if (index == maxLines - 1 && length > 0) {
```

Comment:
I don't believe we want this check on the outer `if`. Consider the example `1\n2\n\n4` with `maxLines = 3`. The expected result would be `1\n2\n...` but the actual result will be `1\n2\n` with no indication that the text was shortened.
Furthermore, I don't understand why `length` is initialized to `contents.lastIndex` rather than `contents.length`. The end of `substring` is exclusive, so if we want to start with the full string, we should start with `length`, not `lastIndex`. That would also fix the (imo) incorrect result this method would give for `1\n2\n3\n4`: expected would be `1\n2\n3...` but, if I'm not mistaken, the actual result will be `1\n2\n...` because we incorrectly start with `lastIndex`, i.e. `0` in the example.